x-pack/plugins/elastic_assistant/server/lib/data_stream/documents_data_writer.ts

```diff
@@ -71,6 +71,10 @@ export class DocumentsDataWriter implements DocumentsDataWriter {
         {
           refresh: 'wait_for',
           body: await this.buildBulkOperations(params),
+          // TODO: Plumb pipeline for update operations so that embeddings are regenerated on update
+          // Changing to `final_pipeline` should work as well according to the docs, see `pipeline` param:
+          // https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-update-by-query.html#docs-update-by-query-api-query-params
+          pipeline: '.kibana-elastic-ai-assistant-ingest-pipeline-knowledge-base',
```

Comment:
Note: Removing this as it did not work as a fix. Passing `pipeline` here didn't regenerate embeddings, which should've worked. Perhaps our painless scripting for managing patch operations is getting in the way here? O11y [has this working](https://github.com/elastic/kibana/blob/6ed731cad5e4f06e985c0521873054adfe2d327c/x-pack/plugins/observability_solution/observability_ai_assistant/server/service/knowledge_base_service/index.ts#L634) with just an `index()` operation, so we may have to do that.

Also note that using a `final_pipeline` didn't work, and caused ELSER/mapping errors on KB initialization, so that's not a solution either.